crypto/evp/digest.c

```diff
@@ -454,6 +454,8 @@ int EVP_DigestFinal_ex(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *isize)
     if (ctx->digest->prov == NULL)
         goto legacy;
 
+    if (sz == 0) /* Assuming a xoflen must have been set. */
```

Comment:
Is it possible that something could have returned 0 for other algorithms?

Reply:
Then it would be an useless hash if it wasn't a XOF one in the same way as SHAKE is.

Alternative is to use the SIZE_MAX as the _invalid default size_ indicator. It has its downsides as well.